derive/src/compile_bytecode.rs

```diff
@@ -115,9 +156,20 @@ impl CompilationSource {
         mode: compile::Mode,
     ) -> Result<HashMap<String, FrozenModule>, Diagnostic> {
         let mut code_map = HashMap::new();
-        let paths = fs::read_dir(&path).map_err(|err| {
-            Diagnostic::spans_error(self.span, format!("Error listing dir {:?}: {}", path, err))
-        })?;
+        let paths = fs::read_dir(path)
+            .or_else(|e| {
+                // handle git symlinks - git for windows will make it just a text file containing
+                // the target
+                if cfg!(windows) && is_git_symlink(path) {
```

Comment:
what if the copied code doesn't have `.git`? e.g. downloading tarball or zip from https://github.com/RustPython/RustPython/archive/refs/heads/master.zip

Reply:
Then it does indeed fail.